### PR TITLE
Fix/fix line item max attempt handling

### PR DESCRIPTION
**Pull request by Skyld**
### Fixed
- Fixed assignment attempt handling logic in case no maximum attempts count defined.

### CODE REVIEW

File: src/Lti/Service/GetUserAssignmentLtiRequestService.php [hunk 1]
```diff
@@ -124,15 +124,25 @@ public function getAssignmentLtiRequest(Assignment $assignment): LtiRequest
...
     {
-        if (!in_array($assignment->getState(), [Assignment::STATE_READY, Assignment::STATE_STARTED], true)) {
+        $lineItem = $assignment->getLineItem();
+        $maxAttempts = $lineItem->getMaxAttempts();
+        $attemptsCount = $assignment->getAttemptsCount();
+
+        if (
+            $lineItem->hasMaxAttempts() &&
+            (
+                $attemptsCount > $maxAttempts ||
```
Comment (luisfn) [inline src/Lti/Service/GetUserAssignmentLtiRequestService.php:134]: Is this condition really possible or only for safety?

**Review by luisfn — COMMENTED** [review]
Seems good to me. Just small things

File: src/Lti/Service/GetUserAssignmentLtiRequestService.php (hunk 1 above)
Comment (Skyld) [reply src/Lti/Service/GetUserAssignmentLtiRequestService.php:134]: It's a safety net, yes :) Technically it should not be possible.

**Review by andreluizmachado — APPROVED** [review]
- [x] New code is covered by tests (if applicable)
- [x] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [x] New code is respecting best practices
- [ ] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [x] Pull request title and description are meaningful
- [x] Changelog is updated according to changes (if applicable)
- [x] Documentation is updated according to changes (if applicable)

**Review by uncleempty — APPROVED** [review]

**Pull request merged**